test/e2e-common.sh

```diff
@@ -20,12 +20,34 @@
 E2E_MIN_CLUSTER_NODES=4
 E2E_MAX_CLUSTER_NODES=4
 
+# Change these two knobs to get a different Istio configuration
+ISTIO_VERSION="1.0.6"      # Istio version.
+ISTIO_OPTION="mesh"        # 'mesh' or 'no-mesh'?
+
 # This script provides helper methods to perform cluster actions.
 source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
 
+# Function to choose an Istio version to test with.
+ISTIO_ROOT="./third_party/"
+
+function istio_manifest() {
+  local istio_version="$1"
+  local suffix="$2"
+
+  if [[ -n $suffix ]]; then
+    suffix="-$suffix"
+  fi
+  echo "$ISTIO_ROOT/istio-${istio_version}/istio${suffix}.yaml"
+}
+
 # Current YAMLs used to install Knative Serving.
-INSTALL_ISTIO_CRD_YAML=""
-INSTALL_ISTIO_YAML=""
+INSTALL_ISTIO_CRD_YAML="$(istio_manifest $ISTIO_VERSION crds)"
+if [[ "$ISTIO_OPTION" == "mesh" ]]; then
```

Comment:
Why an environment variable and not a flag for the istio type?

Reply:
@adrcunha how do I add a test flag for Istio type?  I've used env var here just because I don't know a better way, like test flags.

Let me know the best place to add these options and I'll move them there.  Thanks

Reply:
refactored to flags